Sources/MeiliSearch/Request.swift

```diff
@@ -28,6 +36,29 @@ public final class Request {
     self.session = config.session ?? URLSession.shared
   }
 
+  private func request(
+    _ httpMethod: HTTPMethod,
+    _ url: URL,
+    _ headers: [String: String] = [:],
+    data: Data? = nil
+  ) -> URLRequest {
+    var request = URLRequest(url: url)
```

Comment:
```swift
    var request = URLRequest(url: url)
    request.httpMethod = httpMethod.rawValue
    request.setValue(PackageVersion.qualifiedVersion(), forHTTPHeaderField: "User-Agent")
    
    if httpMethod != .get {
      request.httpBody = data
      request.setValue("application/json", forHTTPHeaderField: "Content-Type")
    }
    
    if let apiKey: String = config.apiKey {
      let bearer = "Bearer \(apiKey)"
      request.setValue(bearer, forHTTPHeaderField: "Authorization")
    }
    
    headers.forEach { key, value in
      request.addValue(value, forHTTPHeaderField: key)
    }

    return request
```

I added some spaces between functions, I think they were important (it is easier to read :D).
But I changed some function execution orders as well, this will help anyone to apply custom data from the `headers` param when needed (idk, if to fully apply this concept we need to switch over `setValue` instead of `addValue`).